### PR TITLE
Spring cleanup

### PR DESCRIPTION
**Pull request by hanzei**
This can be reviewed best via the individual commits.

### CODE REVIEW

File: server/store/kvstore/poll.go [hunk 1]
```diff
@@ -25,16 +27,12 @@ func (s *PollStore) Get(id string) (*poll.Poll, error) {
 	return poll, nil
 }
 
+// Save stores a poll in the KV Store. Overwrittes any existing poll with the same id.
 func (s *PollStore) Save(poll *poll.Poll) error {
-	if err := s.api.KVSet(pollPrefix+poll.ID, poll.EncodeToByte()); err != nil {
-		return err
-	}
-	return nil
+	return s.api.KVSet(pollPrefix+poll.ID, poll.EncodeToByte())
```
Comment (kaakaa) [inline server/store/kvstore/poll.go]: At first, I want to confirm a thing. Can you create a poll by using plugin including this fix?

I encountered nil pointer exception.
```
2018-11-15T00:33:52.141+0900	debug	go-plugin/client.go:823	panic: runtime error: invalid memory address or nil pointer dereference	{"plugin_id": "com.github.matterpoll.matterpoll"}
2018-11-15T00:33:52.141+0900	debug	go-plugin/client.go:823	[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0x15215c6]	{"plugin_id": "com.github.matterpoll.matterpoll"}
2018-11-15T00:33:52.141+0900	debug	go-plugin/client.go:823	{"plugin_id": "com.github.matterpoll.matterpoll"}
2018-11-15T00:33:52.141+0900	debug	go-plugin/client.go:823	goroutine 24 [running]:	{"plugin_id": "com.github.matterpoll.matterpoll"}
2018-11-15T00:33:52.141+0900	debug	go-plugin/client.go:823	github.com/matterpoll/matterpoll/server/vendor/github.com/mattermost/mattermost-server/model.(*AppError).Error(0x0, 0xc00006d8c0, 0x1834dc0)	{"plugin_id": "com.github.matterpoll.matterpoll"}
2018-11-15T00:33:52.141+0900	debug	go-plugin/client.go:823	/Users/nemotoyuusuke/go/src/github.com/matterpoll/matterpoll/server/vendor/github.com/mattermost/mattermost-server/model/utils.go:57 +0x26	{"plugin_id": "com.github.matterpoll.matterpoll"}
2018-11-15T00:33:52.141+0900	debug	go-plugin/client.go:823	github.com/matterpoll/matterpoll/server/plugin.(*MatterpollPlugin).ExecuteCommand(0xc00006c8a0, 0xc0002fae20, 0xc0002fca20, 0x2aad1b0, 0xc00006c8a0)	{"plugin_id": "com.github.matterpoll.matterpoll"}
2018-11-15T00:33:52.141+0900	debug	go-plugin/client.go:823	/Users/nemotoyuusuke/go/src/github.com/matterpoll/matterpoll/server/plugin/command.go:65 +0x3f7	{"plugin_id": "com.github.matterpoll.matterpoll"}
2018-11-15T00:33:52.141+0900	debug	go-plugin/client.go:823	github.com/matterpoll/matterpoll/server/vendor/github.com/mattermost/mattermost-server/plugin.(*hooksRPCServer).ExecuteCommand(0xc00000a640, 0xc0002fab50, 0xc0002fae80, 0x0, 0x0)	{"plugin_id": "com.github.matterpoll.matterpoll"}
2018-11-15T00:33:52.141+0900	debug	go-plugin/client.go:823	/Users/nemotoyuusuke/go/src/github.com/matterpoll/matterpoll/server/vendor/github.com/mattermost/mattermost-server/plugin/client_rpc_generated.go:114 +0x7b	{"plugin_id": "com.github.matterpoll.matterpoll"}
2018-11-15T00:33:52.141+0900	debug	go-plugin/client.go:823	reflect.Value.call(0xc000096ae0, 0xc000094568, 0x13, 0x175e39b, 0x4, 0xc000089f18, 0x3, 0x3, 0xc000133ac0, 0x0, ...)	{"plugin_id": "com.github.matterpoll.matterpoll"}
2018-11-15T00:33:52.141+0900	debug	go-plugin/client.go:823	/usr/local/go/src/reflect/value.go:447 +0x449	{"plugin_id": "com.github.matterpoll.matterpoll"}
2018-11-15T00:33:52.141+0900	debug	go-plugin/client.go:823	reflect.Value.Call(0xc000096ae0, 0xc000094568, 0x13, 0xc000039f18, 0x3, 0x3, 0x0, 0x0, 0x0)	{"plugin_id": "com.github.matterpoll.matterpoll"}
2018-11-15T00:33:52.142+0900	debug	go-plugin/client.go:823	/usr/local/go/src/reflect/value.go:308 +0xa4	{"plugin_id": "com.github.matterpoll.matterpoll"}
2018-11-15T00:33:52.142+0900	debug	go-plugin/client.go:823	net/rpc.(*service).call(0xc00008d700, 0xc0000a06e0, 0xc00008a7e8, 0xc00008a810, 0xc00010c300, 0xc00007b5c0, 0x1627260, 0xc0002fab50, 0x16, 0x16272a0, ...)	{"plugin_id": "com.github.matterpoll.matterpoll"}
2018-11-15T00:33:52.142+0900	debug	go-plugin/client.go:823	/usr/local/go/src/net/rpc/server.go:384 +0x14e	{"plugin_id": "com.github.matterpoll.matterpoll"}
2018-11-15T00:33:52.142+0900	debug	go-plugin/client.go:823	created by net/rpc.(*Server).ServeCodec	{"plugin_id": "com.github.matterpoll.matterpoll"}
2018-11-15T00:33:52.142+0900	debug	go-plugin/client.go:823	/usr/local/go/src/net/rpc/server.go:481 +0x47e	{"plugin_id": "com.github.matterpoll.matterpoll"}
2018-11-15T00:33:52.143+0900	error	plugin/client_rpc_generated.go:104	RPC call ExecuteCommand to plugin failed.	{"plugin_id": "com.github.matterpoll.matterpoll", "error": "unexpected EOF"}
2018-11-15T00:33:52.143+0900	debug	go-plugin/client.go:571	plugin process exited	{"plugin_id": "com.github.matterpoll.matterpoll", "wrapped_extras": "pathplugins/com.github.matterpoll.matterpoll/server/dist/plugin-darwin-amd64"}
2018-11-15T00:33:52.143+0900	error	app/server.go:62	Please check the std error output for the stack trace
2018-11-15T00:33:52.143+0900	error	app/server.go:63	runtime error: invalid memory address or nil pointer dereference
```

If you encountered same error, I think that this changes caused it. We must revert this changes.
(If you can create poll successfully, please let me know it. I'll investigate why error occurs in my machine.)

When `KVSet`/`KVDelete` returns nil, the return value of `Save` func will be nil pointer of `model.AppError` type. Unfortunately, that nil pointer is not nil, so we can't handle it. 

Simple reproduction code is here: https://play.golang.org/p/g7SUC8dF3gN
And please refer [Frequently Asked Questions \(FAQ\) \- The Go Programming Language](https://golang.org/doc/faq#nil_error)

This is weird part of golang.

Comment (hanzei) [reply server/store/kvstore/poll.go]: Oh, you are 100% correct. My bad for not testing.

> This is weird part of golang.

Indeed

**Comment by hanzei** [comment]
@kaakaa Can I merge this?

**Comment by kaakaa** [comment]
Yes,we can

**Pull request merged**